src/components/NewCommentForm.tsx

```diff
@@ -47,25 +66,30 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
+    const newErrors = {
       name: !name,
       email: !email,
       body: !body,
-    });
+      submit: false,
+    };
 
-    if (!name || !email || !body) {
+    setErrors(newErrors);
+
+    if (newErrors.name || newErrors.email || newErrors.body) {
```

Comment:
You are checking if the fields are empty here, but you are not checking if the email is in the correct format. You should add a regex check here to ensure that the email is a valid email address.